sql/src/test/java/io/crate/analyze/AlterTableRerouteAnalyzerTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+
+public class AlterTableRerouteAnalyzerTest extends CrateDummyClusterServiceUnitTest {
+
+    private SQLExecutor e;
+
+    @Before
+    public void prepare() {
+        e = SQLExecutor.builder(clusterService).enableDefaultTables().build();
+    }
+
+    @Test
+    public void testRerouteOnSystemTableIsNotAllowed() throws Exception {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("The relation \"sys.cluster\" doesn't support or allow ALTER operations, as it is read-only.");
+        e.analyze("ALTER TABLE sys.cluster REROUTE RETRY FAILED");
+    }
+
+    @Test
+    public void testRerouteRetryOnUnpartitionedTableWithPartitionIsNotAllowed() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("table 'doc.users' is not partitioned");
+        e.analyze("ALTER TABLE doc.users PARTITION (col_a = NULL) REROUTE RETRY FAILED");
```

Comment:
this is irrelevant to your new feature. as it seems to be not tested, pls add a PartitionPropertiesAnalyzerTest unit test instead